### PR TITLE
Profile Set Username

### PR DESCRIPTION
**Pull request by sev3rus**
Set username in profile
Also fetch username in dashboard screen from datastore

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -164,6 +164,7 @@
     <string name="label_setting_privacy_policy">Privacy Policy</string>
     <string name="label_setting_privacy_policy_description">Click to see our privacy policy</string>
     <string name="label_setting_app_version">App Version : </string>
+    <string name="name">Name</string>
```
Comment (Audienix) [inline app/src/main/res/values/strings.xml]: label_setting_name is already there, can't we use it?

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt [hunk 2]
```diff
@@ -169,3 +184,79 @@ fun AppVersion() {
...
+    textInputAttributes: TextInputAttributes,
+    onCancelClick: () -> Unit,
+    onConfirmClick: () -> Unit
+) {
+    AlertDialog(
+        onDismissRequest = {}) {
+        Surface(
+            shape = Shapes.extraLarge,
+            color = MaterialColorPalette.surfaceContainerHigh,
+            tonalElevation = dimensionResource(id = R.dimen.dimension_6dp),
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt]: do we require this?

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt [hunk 3]
```diff
@@ -169,3 +184,79 @@ fun AppVersion() {
...
+            labelTextId = R.string.name
+        ),
+        onCancelClick =  { viewModel.setAction(null, "")},
+        onConfirmClick = { viewModel.setName(viewModel.currentPopupValue) }
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun IPTextInputDialog(
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt]: move this under package `com.twain.interprep.presentation.ui.components.generic` ?

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt [hunk 4]
```diff
@@ -169,3 +184,79 @@ fun AppVersion() {
...
+                IPTextInput(
+                    inputText = inputText,
+                    textInputAttributes = textInputAttributes,
+                    onTextUpdate = onTextUpdate
+                )
+                Spacer(modifier = Modifier.height(dimensionResource(id = R.dimen.dimension_20dp)))
+                Row(
+                ) {
+                    Spacer(modifier = Modifier.weight(1f))
+                    TextButton(onClick = onCancelClick) {
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt]: Add  
colors = ButtonDefaults.buttonColors(
                    containerColor = Color.Transparent,
                    contentColor = MaterialColorPalette.primary
                )

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt [hunk 5]
```diff
@@ -169,3 +184,79 @@ fun AppVersion() {
...
+                        Text(text = "Cancel")
+                    }
+                    TextButton(onClick = onConfirmClick) {
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt]: Add
colors = ButtonDefaults.buttonColors(
containerColor = Color.Transparent,
contentColor = MaterialColorPalette.primary
)

**Review by Audienix — APPROVED** [review]
few minor comments to address before merging.

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt (hunk 2 above)
Comment (sev3rus) [reply app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt]: This specs are same as Compose default Alertdialog, just copied every token from that component

**Pull request merged**